app/routes.py

```diff
@@ -1,2 +1,179 @@
+
+@tasks_bp.route("", methods=["POST"])
+def post_tasks():
+    request_body = request.get_json()
+    if "title" in request_body.keys() and "description" in request_body.keys() and "completed_at" in request_body.keys() :
+        task = Task(title=request_body["title"],
+                    description=request_body["description"], 
+                    completed_at=request_body["completed_at"],
+                    )
+        db.session.add(task)
+        db.session.commit()
+        return jsonify({"task": task.api_response()}), 201
+    else:
+        return make_response(
+            {"details": "Invalid data"
+            }
+        ), 400
+
+@tasks_bp.route("", methods=["GET"])
+def get_tasks():
+    sort_param = request.args.get("sort")
+    if sort_param == "asc":
+        tasks = Task.query.order_by(Task.title).all()
+    elif sort_param == "desc":
+        tasks = Task.query.order_by(Task.title.desc()).all()
+    else:
+        tasks = Task.query.all()
+    tasks_response = [task.api_response() for task in tasks] 
+    return jsonify(tasks_response), 200
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def get_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response(jsonify(None), 404)
+    return jsonify({"task": task.api_response()}), 200
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def put_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return Response(None),404
+    form_data = request.get_json()
+    task.title = form_data["title"]
+    task.description = form_data["description"]
+    task.completed_at = form_data["completed_at"]
+    db.session.commit()
+    return jsonify({"task": task.api_response()}), 200 
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return Response(None),404
+    task.completed_at = datetime.now()
+    db.session.commit()
+    slack_bot_complete(task.title)
+    return jsonify({"task": task.api_response()}), 200   #removed "true" as argument
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return Response(None),404
+    task.completed_at = None
+    db.session.commit()
+    return jsonify({"task": task.api_response()}), 200    
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return Response(None),404
+    db.session.delete(task)
+    db.session.commit()
+    return make_response(
+        {"details": f'Task {task.id} "{task.title}" successfully deleted'
+        }
+    ), 200
+
+@goals_bp.route("", methods=["POST"])
+def post_goals():
+    request_body = request.get_json()
+    if "title" in request_body.keys():
+        goal = Goal(title=request_body["title"],
+                    )
```

Comment:
careful with trailing commas! sometimes they can turn value into tuples or cause errors
```suggestion
        goal = Goal(title=request_body["title"])
```